File: C++/4.Arrays & String/3.Sorting/insertionSort.cpp
#include <iostream>
using namespace std;

void insertionSort (int a[], int n) {


}

//print sorted array element
void printArray(int a[], int n) {
    for (int i = 0; i < n; i++) {
        cout << a[i] << " ";
    } cout << endl;    
}

int main() {
    
    int arr[] = {5, 4, 2, 6, 3, 9, 7, 4, 5, 2};

    int size = sizeof(arr) / sizeof (arr[0]);

    cout <<"insertion Sort: "<<endl;
    insertionSort (arr, size);
    printArray(arr, size);

    return 0;
}

File: Java/3.Arrays/sorting/InsertionSort.java
public class InsertionSort {
    public static void main(String[] args) {

        System.out.println("Insertion Sort: ");
        
        int[] arr = {7, 4, 1, 2, 5, 8, 9, 6, 3};
        int n = arr.length;
        for (int i = 1; i < n; i++) { // total round -> start from the second element assume the first elment is sorted
            int key = arr[i]; //store second element to the key
            int j = i-1;
            for (; j >= 0; j--) { // this loop is for compare key element to its previous adjecent element
                if (arr[j] > key)
                    arr[j+1] = arr[j]; // shift to the right
                else
                    break;
            }
            arr[j+1] = key;
        }

        // print the sorted element
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }
    }
}
